src/devices/Nmea0183/tests/NetworkPortScanner.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Net;
+using System.Text;
+
+namespace Iot.Device.Nmea0183.Tests
+{
+    /// <summary>
+    /// This class helps finding a free network port for a test.
+    /// Could be moved to the real library, but with what name?
+    /// </summary>
+    internal class NetworkPortScanner
```

Comment:
I'd really prefer we did not have to use TCP or UDP anywhere in the product and not have to test it this way

Reply:
(these kind of stuff tends to be flaky and networking code is always a bit fragile and prone to errors)

Reply:
I know, but this is the best possibility to really do an integration test on several parts at once. And it _did_ help me find compatibility bugs on different platforms (e.g. macOS has several weird bugs in the UDP code).